src/api/artifact/controller.go

```diff
@@ -476,10 +481,61 @@ func (c *controller) ListTags(ctx context.Context, query *q.Query, option *TagOp
 func (c *controller) DeleteTag(ctx context.Context, tagID int64) error {
 	// Immutable checking is covered in middleware
 	// TODO check signature
-	// TODO fire delete tag event
+	c.sendTagDeleteEvent(ctx, tagID)
```

Comment:
We support deleting tag and artifact in Harbor 2.0: 
- all tags attached to the artifact will be deleted when the artifact is deleted
- a tag can be deleted without deleting the artifact
That means deleting tag and deleting artifact are two separated event